packages/wrangler/src/api/pages/publish.tsx

```diff
@@ -243,13 +248,55 @@ export async function publish({
 	 * When using a _worker.js file, the entire /functions directory is ignored
 	 * – this includes its routing and middleware characteristics.
 	 */
-	if (_workerJS) {
+	if (_workerJSDirectory) {
+		const traverseModuleGraphResult = await traverseModuleGraph(
+			{
+				file: resolvePath(join(_workerPath, "index.js")),
+				directory: resolvePath(_workerPath),
+				format: "modules",
+				moduleRoot: resolvePath(_workerPath),
+			},
+			[
+				{
+					type: "ESModule",
+					globs: ["**/*.js"],
+				},
+			]
+		);
+
+		const outfile = join(tmpdir(), `./bundledWorker-${Math.random()}.mjs`);
+		const bundleResult = await buildRawWorker({
+			workerScriptPath: _workerPath,
+			bundle: false,
+			outfile,
+			directory,
+			local: false,
+			sourcemap: true,
+			watch: false,
+			onEnd: () => {},
+			betaD1Shims: d1Databases,
+			nodejsCompat,
+		});
+
+		workerBundle = {
+			modules: (traverseModuleGraphResult?.modules ??
+				bundleResult?.modules) as BundleResult["modules"],
+			dependencies: (bundleResult?.dependencies ??
+				traverseModuleGraphResult?.dependencies) as BundleResult["dependencies"],
+			resolvedEntryPointPath: (bundleResult?.resolvedEntryPointPath ??
+				traverseModuleGraphResult?.resolvedEntryPointPath) as BundleResult["resolvedEntryPointPath"],
+			bundleType: (bundleResult?.bundleType ??
+				traverseModuleGraphResult?.bundleType) as BundleResult["bundleType"],
```

Comment:
I'm confused by what this is doing. Why does `traverseModuleGraphResult` take precedence in some cases, and `bundleResult` in others?

Reply:
Discussed IRL. Agreed that a combination of bits from both makes sense sense. For example, `bundleWorker` needs its watch stuff, and `traverseModuleGraph` gives a set of modules.